interfaces/builtin/opengl.go

```diff
@@ -44,12 +44,16 @@ const openglConnectedPlugAppArmor = `
 /var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libGLdispatch.so{,.*} rm,
 /var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}vdpau/libvdpau_nvidia.so{,.*} rm,
 
-# Support reading the Vulkan ICD files
+# Support reading the Vulkan ICD and layer files
+/usr/share/vulkan/icd.d/{,*.json} r,
+/usr/share/vulkan/explicit_layer.d/{,*.json} r,
+/usr/share/vulkan/implicit_layer.d/{,*.json} r,
 /var/lib/snapd/lib/vulkan/ r,
 /var/lib/snapd/lib/vulkan/** r,
 /var/lib/snapd/hostfs/usr/share/vulkan/icd.d/*nvidia*.json r,
```

Comment:
Wonder whether for symmetry we should allow `*.json` from hostfs. @zyga given your experiments, does that make sense?

Reply:
The reason the hostfs read access is limited to `*nvidia*.json` is because that is (currently) the only hostfs driver linked to `/var/lib/snapd/lib/vulkan`.  Unless snap-confine was enhanced to expose more host drivers to the sandbox, it doesn't really help much for the sandboxed app to know about other host drivers.  And in the specific case of the Mesa/DRI drivers, doing so is particularly difficult due to C++ ABI differences (at least if Mesa is compiled against a newer libstdc++ than exists in the base snap).

So while the two cases are similar, I don't think I'd describe them as symmetric.

Reply:
I agree with @jhenstridge

Reply:
What I meant is that s-c already cherry pick nvidia files only, so the AA rules for `egl_vendor.d` and Vulkan's `icd.d` may actually be generic and look like the ones for Mesa

Reply:
@bboozzoo I think we should refrain from reading configuration files from hostfs. The direction we have set is clear: sharing /etc was a mistake we will eventually undo. We need a layer of control and that only works if snapd can re-interpret applicable host configuration and inject it into the snap world. Therefore I would be -1 on sharing additional config files from hostfs this way.

Reply:
> The direction we have set is clear: sharing /etc was a mistake we will eventually undo.

I think this is overstated and not representative of everyone's opinion or considerate of the historical context. There were good reasons to use /etc from the host when the decision. We of course may want to revisit this or what to selectively expose, but that discussion can happen elsewhere.